test/lit/passes/optimize-casts.wast

```diff
@@ -387,6 +412,282 @@
     )
   )
 
+  ;; CHECK:      (func $move-cast (type $ref|struct|_=>_none) (param $x (ref struct))
+  ;; CHECK-NEXT:  (local $1 (ref $B))
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (ref.cast $B
+  ;; CHECK-NEXT:    (local.tee $1
+  ;; CHECK-NEXT:     (ref.cast $B
+  ;; CHECK-NEXT:      (local.get $x)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (local.get $1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (ref.cast $B
+  ;; CHECK-NEXT:    (local.get $1)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $move-cast (param $x (ref struct))
+    (drop
```

Comment:
An explanation of what to expect would be good - it really helps speed up reading a testcase when you come back to it later and don't remember all the context.

Here, for example, I'd write
```
The most refined cast of $x is to $B, which we can move up to
the top and reuse from there.
```
And to address my question from earlier, which a future reader might also have, can add "(the cast to $A will also turn into a cast to $B due to ReFinalize)".

Reply:
Done.